omnigibson/object_states/particle_modifier.py

```diff
@@ -194,18 +194,18 @@ def __init__(self, obj, method, conditions, projection_mesh_params=None):
         self._projection_mesh_params = projection_mesh_params
 
         # Map of system to number of modified particles (only include systems specified in the conditions)
-        self.modified_particle_count = dict([(system, 0) for system in self.conditions])
+        self.modified_particle_count = dict([(system_name, 0) for system_name in self.conditions])
 
         # Standardize the conditions (make sure every system has at least one condition, which to make sure
         # the particle modifier isn't already limited with the specific number of particles)
-        for system, conds in conditions.items():
+        for system_name, conds in conditions.items():
             # Make sure the system is supported
-            assert_valid_key(key=system, valid_keys=ParticleModifier.supported_systems, name="particle system")
+            assert is_visual_particle_system(system_name) or is_physical_particle_system(system_name), f"Unsupported system for ParticleModifier {system_name}"
             # Make sure conds isn't empty and is a list
             conds = [] if conds is None else list(conds)
             # Add the condition to avoid limits
-            conds.append(self._generate_limit_condition(system=system))
-            conditions[system] = conds
+            conds.append(self._generate_limit_condition(system_name))
```

Comment:
can we update the docstring for this class so it's clear we're mapping strings to conditions now instead of classes to conditions?

Reply:
done